content/courses/mobile/solana-mobile-dapps-with-expo.md

```diff
@@ -989,85 +1200,93 @@ through the code for each of them and then show you the entire file at the end:
    We'll create two helper functions for uploading the image and metadata
    separately, then tie them together into a single `createNFT` function:
 
-   ```tsx
-   // https://nft.storage/api-docs/
-   const uploadImage = async (fileUri: string): Promise<string> => {
-     const imageBytesInBase64: string = await RNFetchBlob.fs.readFile(
-       fileUri,
-       "base64",
-     );
-     const bytes = Buffer.from(imageBytesInBase64, "base64");
-
-     const response = await fetch("https://api.nft.storage/upload", {
-       method: "POST",
-       headers: {
-         Authorization: `Bearer ${process.env.EXPO_PUBLIC_NFT_STORAGE_API}`,
-         "Content-Type": "image/jpg",
-       },
-       body: bytes,
-     });
-
-     const data = await response.json();
-     const cid = data.value.cid;
-
-     return cid as string;
-   };
-
-   const uploadMetadata = async (
-     name: string,
-     description: string,
-     imageCID: string,
-   ): Promise<string> => {
-     const response = await fetch("https://api.nft.storage/upload", {
-       method: "POST",
-       headers: {
-         Authorization: `Bearer ${process.env.EXPO_PUBLIC_NFT_STORAGE_API}`,
-       },
-       body: JSON.stringify({
-         name,
-         description,
-         image: `https://ipfs.io/ipfs/${imageCID}`,
-       }),
-     });
-
-     const data = await response.json();
-     const cid = data.value.cid;
-
-     return cid;
-   };
-   ```
-
-   Minting the NFT after the image and metadata have been uploaded is as simple
-   as calling `metaplex.nfts().create(...)`. Below shows the `createNFT`
-   function tying everything together:
-
-   ```tsx
-   const createNFT = async (
-     name: string,
-     description: string,
-     fileUri: string,
-   ) => {
-     if (!metaplex || !account || isLoading) return;
-
-     setIsLoading(true);
-     try {
-       const imageCID = await uploadImage(fileUri);
-       const metadataCID = await uploadMetadata(name, description, imageCID);
-
-       const nft = await metaplex.nfts().create({
-         uri: `https://ipfs.io/ipfs/${metadataCID}`,
-         name: name,
-         sellerFeeBasisPoints: 0,
-       });
-
-       setNftOfTheDay(nft.nft);
-     } catch (error) {
-       console.log(error);
-     } finally {
-       setIsLoading(false);
-     }
-   };
-   ```
+```tsx
+// https://nft.storage/api-docs/
+const uploadImage = async (fileUri: string): Promise<string | undefined> => {
+  try {
+    const imageBytesInBase64: string = await RNFetchBlob.fs.readFile(
```

Comment:
Where is RNFetchBlob imported from?

Reply:
It was imported from rn-fetch-blob npm package but not needed anymore since nft.storage has shifted their architecture. Leveraging Pinata will be better for the developer experience and doesn't need data as blobs.